packages/realm/src/TypeHelpers.ts

```diff
@@ -30,22 +30,26 @@ import {
   UpdateMode,
   assert,
   binding,
+  safeGlobalThis,
 } from "./internal";
 
-const TYPED_ARRAY_CONSTRUCTORS = new Set([
-  DataView,
-  Int8Array,
-  Uint8Array,
-  Uint8ClampedArray,
-  Int16Array,
-  Uint16Array,
-  Int32Array,
-  Uint32Array,
-  Float32Array,
-  Float64Array,
-  BigInt64Array,
-  BigUint64Array,
-]);
+const TYPED_ARRAY_CONSTRUCTORS = new Set(
+  [
+    DataView,
+    Int8Array,
+    Uint8Array,
+    Uint8ClampedArray,
+    Int16Array,
+    Uint16Array,
+    Int32Array,
+    Uint32Array,
+    Float32Array,
+    Float64Array,
+    // These will not be present on old versions of JSC without BigInt support.
+    safeGlobalThis.BigInt64Array,
+    safeGlobalThis.BigUint64Array,
+  ].filter((ctor) => ctor !== undefined),
```

Comment:
Since they're added to a set, I wonder if it's even worth bothering to filter?

Reply:
Without the filter, you'd have `undefined` in the set on JSC, which would be an issue in https://github.com/realm/realm-js/blob/99d3fee8338885e00b5e6cb8120191d468b61437/packages/realm/src/TypeHelpers.ts#L58-L61